commands/invoke.go

```diff
@@ -116,5 +115,12 @@ func (cl *invokeCmd) Invoke(c *cli.Context) error {
 		}
 	}
 
-	return client.Invoke(cl.provider, invokeURL, content, os.Stdout, c.String("method"), c.StringSlice("e"), contentType, c.Bool("display-call-id"))
+	err := client.Invoke(cl.provider, invokeURL, content, os.Stdout, c.String("method"), c.StringSlice("e"), contentType, c.Bool("display-call-id"))
+	// we don't want to show the help message if invoke fails, just copy error to
+	// stderr but don't return the error from Invoke. also exit early here
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err.Error())
+		os.Exit(1)
```

Comment:
are there any thoughts about this vs getting rid of https://github.com/fnproject/cli/blob/master/main.go#L246-L249 ?

for invoke specifically I think it's really weird for us to garble up the output of users functions that don't really error in the context of fn (the function runs) with client version and all that, but I'm not opposed to getting rid of all that output altogether (my cli philosophy usually doesn't align with the corporate ones, however, so leaving this open...)

Reply:
is the behavior here that if the invoke fails, we should the Fn:, Client Version etc? If so, I don't think we should be showing that.

Reply:
this has improved with https://github.com/fnproject/cli/pull/538/commits/ca67a7b284d3de29190c69cbd39445fab667a792 and this block has gone away. now we only error and show the crap if there's issues building the request or talking to the service [without getting a response], and it's more clear what we do with output regardless of error status